python/packages/autogen-ext/src/autogen_ext/agents/azure/_azure_ai_agent.py

```diff
@@ -0,0 +1,986 @@
+                        project_client=project_client,
+                        deployment_name="gpt-4o",
+                        instructions="You are a helpful assistant.",
+                        tools=["file_search"],
+                        metadata={"source": "AzureAIAgent"},
+                    )
+
+                    await agent_with_file_search.on_upload_for_file_search(
+                        file_paths=[
+                            "/workspaces/autogen/python/packages/autogen-core/docs/src/user-guide/core-user-guide/cookbook/data/product_info_1.md"
+                        ],
+                        vector_store_name="file_upload_index",
+                        vector_store_metadata={"source": "AzureAIAgent"},
+                        cancellation_token=CancellationToken(),
+                    )
+                    result = await agent_with_file_search.on_messages(
+                        messages=[TextMessage(content="Hello, what Contoso products do you know?", source="user")],
+                        cancellation_token=CancellationToken(),
+                        message_limit=5,
+                    )
+                    print(result)
+
+                if __name__ == "__main__":
+                    dotenv.load_dotenv()
+                    asyncio.run(file_search_example())
+
+        Use the AzureAIAgent to create an agent with code interpreter capability:
+
+        .. code-block:: python
+
+            import asyncio
+            import os
+
+            import dotenv
+            from autogen_agentchat.messages import TextMessage
+            from autogen_core import CancellationToken
+            from autogen_ext.agents.azure._azure_ai_agent import AzureAIAgent
+            from azure.ai.projects.aio import AIProjectClient
+            from azure.identity.aio import DefaultAzureCredential
+
+
+            async def code_interpreter_example():
+                credential = DefaultAzureCredential()
+                async with AIProjectClient.from_connection_string(
+                    credential=credential, conn_str=os.getenv("AI_PROJECT_CONNECTION_STRING", "")
+                ) as project_client:
+                    agent_with_code_interpreter = AzureAIAgent(
+                        name="code_interpreter_agent",
+                        description="An AI assistant with code interpreter capabilities",
+                        project_client=project_client,
+                        deployment_name="gpt-4o",
+                        instructions="You are a helpful assistant.",
+                        tools=["code_interpreter"],
+                        metadata={"source": "AzureAIAgent"},
+                    )
+
+                    await agent_with_code_interpreter.on_upload_for_code_interpreter(
+                        file_paths="/workspaces/autogen/python/packages/autogen-core/docs/src/user-guide/core-user-guide/cookbook/data/nifty_500_quarterly_results.csv",
```

Comment:
For code interpreter, we can use CSV file in public data. For example: https://data.seattle.gov/City-Administration/City-of-Seattle-Wage-Data/2khk-5ukd/about_data.